topics/reference_guide/custom_language_support/websymbols_integration.md

```diff
@@ -0,0 +1,309 @@
+<!-- Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+# Web Symbols Integration with language features
+
+<link-summary>
+How to integrate your Web Symbols with a particular language feature.
+</link-summary>
+
+IDEs provide built-in support for Web Symbols integration with main language features of HTML, CSS and JavaScript.
+You can contribute static symbols through Web Types or Custom Elements Manifest, or you can
+create and register [`WebSymbolsQueryConfigurator`](%gh-ic%/platform/webSymbols/src/com/intellij/webSymbols/query/WebSymbolsQueryConfigurator.kt)
+extension.
+
+Your implementation of `WebSymbolsQueryConfigurator` needs to override `getScope` method, in which you
+create a list of `WebSymbolScope`s depending on the provided `PsiElement` (`element` parameter) and `WebSymbolContext` (`context` parameter).
+The below list of supported language features specify information on what kind of `PsiElement`s to expect for each of
+the supported language features.
+
+## Supported language features
+
+### HTML
+
+#### Elements {#html-elements}
+
+|---|---|
+|||
+|Namespace|`html`|
+|Kind|`elements`|
+
+Web Symbols, which represent available HTML elements.
+HTML element symbols can be contributed statically or dynamically (through `WebSymbolsQueryConfigurator`).
+The statically contributed symbols are available globally, depending on the context setting of the contributing Web Types file.
+
+For dynamic contributions the `WebSymbolsQueryConfigurator#getScope`'s `element` parameter can be:
+- `HtmlTag` - Web Symbols should represent available HTML elements. *The HTML tag actual name should not be taken into account when building the scope.*
+- `CssElement` - Web Symbols should represent available HTML elements within a particular CSS file.
+
+The matched Web Symbol is taken as a scope for matching HTML attributes of the tag.
+
+#### Attributes {#html-attributes}
+
+|---|---|
+|||
+|Namespace|`html`|
+|Kind|`attributes`|
+
+Web Symbols, which represent available HTML attributes.
+HTML attribute symbols can be contributed statically or dynamically (through `WebSymbolsQueryConfigurator`).
+The statically contributed symbols are available globally, depending on the context setting of the contributing Web Types file.
+If the containing tag is matched to a Web Symbol it is added to the scope for attribute matching.
+
+For dynamic contributions the `WebSymbolsQueryConfigurator#getScope`'s `element` parameter can be:
```

Comment:
```suggestion
For dynamic contributions the `WebSymbolsQueryConfigurator.getScope()`'s `element` parameter can be:
```

We don't use `#` to refer to methods. If it is clear from the context, we could also leave out the `WebSymbolsQueryConfigurator` and just use `getScope()`.